R/corpus-methods-quanteda.R

```diff
@@ -84,46 +21,38 @@ metacorpus.corpus <- function(x, field = NULL) {
 #' nchar(texts(corpus_subset(data_corpus_inaugural, Year < 1806), groups = "President"))
 #' 
 #' # grouping a character vector using a factor
-#' nchar(data_char_ukimmig2010[1:5])
 #' nchar(texts(data_corpus_inaugural[1:5], 
-#'             groups = as.factor(data_corpus_inaugural[1:5, "President"])))
+#'       groups = "President"))
+#' nchar(texts(data_corpus_inaugural[1:5], 
+#'       groups = factor(c("W", "W", "A", "J", "J"))))
 #' 
-texts <- function(x, groups = NULL, spacer = "  ") {
+texts <- function(x, groups = NULL, spacer = " ") {
```

Comment:
If it inserts two " ", `corpus_reshape()` would not restore original texts. When do we need two " "?

Reply:
I'm not a stickler for preserving "original texts", and would rather have a structured representation of all original data, but not identical to the original. We don't want to lose actual text, but delimiters ought to be regularized. Why? I'd rather have:
- single spaces between tokens rather than multiple.
- regular old ASCIII 32 for a space, and only preserve `\t` and `\n` outside of that. Who cares about preserving a "FOUR-PER-EM SPACE" ("\u2005")?? I favour normalization.
- two spaces between sentences
- two newlines between paragraphs.